climakitae/uncertain_utils.py

```diff
@@ -0,0 +1,426 @@
+    location
+        geographic area name (name of county/state)
+    timescale
+        frequency of data
+    area_average: bool
+        average computed across domain
+
+    Methods
+    -------
+    cmip_clip
+        CMIP6-specific subsetting
+    """
+
+    def __init__(
+        self,
+        variable="tas",  ## set up for temp uncertainty notebook
+        area_subset="states",
+        location="California",
+        timescale="monthly",
+        area_average=True,
+    ):
+        self.variable = variable
+        self.area_subset = area_subset
+        self.location = location
+        self.area_average = area_average
+        self.timescale = timescale
+
+    def _cmip_clip(self, ds):
+        """CMIP6 function to subset dataset based on the data selection options.
+
+        Parameters
+        ----------
+            ds: xr.Dataset
+                Input data
+
+        Returns
+        -------
+            ds: xr.Dataset
+                Subsetted data, area-weighting applied if area_average is true
+        """
+        variable = self.variable
+        location = self.location
+        area_average = self.area_average
+        area_subset = self.area_subset
+        timescale = self.timescale
+
+        to_drop = [v for v in list(ds.data_vars) if v != variable]
+        ds = ds.drop_vars(to_drop)
+        ds = _clip_region(ds, area_subset, location)
+        if area_average:
+            ds = _area_wgt_average(ds)
+        return ds
+
+
+def _cf_to_dt(ds):
+    """Converts non-standard calendars using cftime to pandas datetime
+
+    Parameters
+    ----------
+        ds: xr.Dataset
+            Input data
+
+    Returns
+    -------
+        ds: xr.Dataset
+            Converted calendar data
+    """
+    if type(ds.indexes["time"]) not in [pd.core.indexes.datetimes.DatetimeIndex]:
+        datetimeindex = ds.indexes["time"].to_datetimeindex()
+        ds["time"] = datetimeindex
+    return ds
+
+
+def _calendar_align(ds):
+    """Aligns calendars for consistent matching
+
+    CMIP6 function to set the day for all monthly values to the the 1st of
+    each month, as models have different calendars (e.g., no leap day, 360 days),
+    which can result in a large dataset with empty values in time when concatenated.
+
+    WARNING: can impact functions which use number of days in each month (e.g.,
+    precipitation flux to total monthly accumulation).
+
+    Parameters
+    ----------
+        ds: xr.Dataset
+            Input data
+
+    Returns
+    -------
+        ds: xr.Dataset
+            Calendar-aligned data
+    """
+    ds["time"] = pd.to_datetime(ds.time.dt.strftime("%Y-%m"))
+    return ds
+
+
+def _clip_region(ds, area_subset, location):
+    """Clips CMIP6 dataset using a polygon.
+
+    Parameters
+    ----------
+        ds: xr.Dataset
+            Input data
+        area_subset: string
+            "counties"/"states" as options
+        location: string
+            county/state name
```

Comment:
This should say ``str`` instead of ``string``. Also, is location a string or is is the LocSelectorArea object?